### PR TITLE
Feature/upgrade js libs

### PR DESCRIPTION
**Pull request by doolse**
Upgrade material ui to 3.9.2, change deprecated typography variants and fix some imports

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { sprintf } from "sprintf-js";
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts]: unused import?

File: Source/Plugins/Core/com.equella.core/js/tsrc/course/SearchCourse.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { courseService } from "../services";
 import { prepLangStrings, formatSize } from "../util/langstrings";
 import VisibilitySensor = require("react-visibility-sensor");
 import EntityList from "../components/EntityList";
+import { sprintf } from "sprintf-js";
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/course/SearchCourse.tsx]: unused import?

File: Source/Plugins/Core/com.equella.core/js/tsrc/course/EditCourse.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ import {
   TextField,
   Theme
 } from "@material-ui/core";
+import { sprintf } from "sprintf-js";
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/course/EditCourse.tsx]: unused import?

File: Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts (hunk 1 above)
Comment (doolse) [reply Source/Plugins/Core/com.equella.core/js/tsrc/util/langstrings.ts]: Nope, somewhere during the upgrade, sprintf became no longer in global scope so had to be imported explicitly. (Not sure how it was ever in global scope actually)

File: Source/Plugins/Core/com.equella.core/js/src/Uploads/UploadModel.purs [hunk 4]
```diff
@@ -1,10 +1,10 @@
 module Uploads.UploadModel where 
 
-import Effect.Aff.Compat
-import Prelude
+import Effect.Aff.Compat (EffectFnAff, fromEffectFnAff)
+import Prelude
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/src/Uploads/UploadModel.purs]: :art: remove trailing whitespace
```suggestion
import Prelude
```

File: Source/Plugins/Core/com.equella.core/js/package.json [hunk 5]
```diff
@@ -24,39 +24,33 @@
...
+    "@date-io/luxon": "^1.1.0",
+    "@material-ui/core": "^3.9.2",
+    "@material-ui/icons": "^3.0.2",
+    "@material-ui/lab": "^3.0.0-alpha.30",
     "axios": "^0.18.0",
     "create-react-class": "^15.6.2",
     "downshift": "^2.2.2",
-    "draft-js": "0.10.5",
-    "draftjs-utils": "0.9.4",
+    "draft-convert": "^2.1.4",
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/package.json]: no longer needed IIRC. ended up using draftjs-to-html and html-to-draftjs for this purpose. Same with draft-js-export-html, that is no longer needed either.

File: Source/Plugins/Core/com.equella.core/js/src/UI/MessageInfo.purs [hunk 6]
```diff
@@ -2,10 +2,9 @@ module OEQ.UI.MessageInfo where
 
 import Prelude
 
-import Data.Nullable (Nullable)
 import Data.TSCompat (OneOf, OptionRecord, StringConst)
 import Data.TSCompat.Class (class IsTSEq)
-import Effect (Effect)
+import Effect (Effect)
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/js/src/UI/MessageInfo.purs]: extraneous space

**Pull request merged**